specification/automation/resource-manager/Microsoft.Automation/stable/2018-01-15/dscNodeCounts.json

```diff
@@ -0,0 +1,149 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "AutomationManagement",
+    "version": "2018-01-15",
+    "x-ms-code-generation-settings": {
+      "useDateTimeOffset": true
+    }
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Automation/automationAccounts/{automationAccountName}/nodecounts/{countType}": {
```

Comment:
why is nodeCounts modeled as a separate resource type? Could it be just a property on the automation Account or on nodes?

Reply:
@ravbhatnagar - Not sure what you mean by "a property on the AutomationAccount or on nodes" but the following was my reasoning for a separate resource... 

/subscriptions/{SubId}/.../nodes specifies getting a list of all nodes and /subscriptions/{SubId}/.../nodes/{NodeId} specifies getting node information for NodeId.  If I were to create a url like this: /subscriptions/{SubId}/.../nodes/StatusCount or /subscriptions/{SubId}/.../nodes/NodeConfigurationCount then any user who tried to create a node with StatusCount or NodeConfigurationCount as a name would break our system.  

The count data is a different view of nodes where getting the status count is essentially a count of all nodes by status (think SQL group by) and  getting the nodeConfiguration count is getting a count of all nodes with a particular node configuration value making the count data not a property of a node but a property of the entire collection of nodes.  I modeled counts as a separate resource due to the above because all other approaches seemed wrong.